dev/TEAL.md

```diff
@@ -286,12 +305,17 @@ Asset fields include `AssetHolding` and `AssetParam` fields that are used in `as
 | `pop` | discard value X from stack |
 | `dup` | duplicate last value on stack |
 | `dup2` | duplicate two last values on stack: A, B -> A, B, A, B |
+| `dig` | duplicate value N from the top of stack |
+| `swap` | swaps two last values on stack: A, B -> B, A |
+| `select` | selects one of two values to retain: A, B, C -> A ? B : C |
+| `assert` | immediately fail unless value X is a non-zero number |
 
 ### State Access
 
 | Op | Description |
 | --- | --- |
 | `balance` | get balance for the requested account specified by Txn.Accounts[A] in microalgos. A is specified as an account index in the Accounts field of the ApplicationCall transaction, zero index means the sender |
+| `min_balance` | get minimum balance for the requested account specified by Txn.Accounts[A] in microalgos. A is specified as an account index in the Accounts field of the ApplicationCall transaction, zero index means the sender |
```

Comment:
What does "minimum balance" mean?

Reply:
The number of algos that the account must retain for a transaction it's involved it to succeed.  It's pretty difficult to know or calculate this amount, since it is affected by opt-ins to ASAs and Apps, with some involved calculations on several consensus values in the case of Apps. (because it depends on schema sizes and kinds)

We found that a uniswap-like AMM needs this operation in order to calculate how many "effective algos" it has under escrow.